pkg/compiler/debug.go

```diff
@@ -594,23 +595,37 @@ func (di *DebugInfo) ConvertToManifest(o *Options) (*manifest.Manifest, error) {
 	if o.ContractSupportedStandards != nil {
 		result.SupportedStandards = o.ContractSupportedStandards
 	}
+	namedTypes := make(map[string]manifest.ExtendedType)
+	for name, et := range o.GuessedNamedTypes {
+		namedTypes[di.MainPkg+"."+name] = et
+	}
+	for name, et := range di.NamedTypes {
+		namedTypes[name] = et
+	}
 	events := make([]manifest.Event, len(o.ContractEvents))
 	for i, e := range o.ContractEvents {
 		params := make([]manifest.Parameter, len(e.Parameters))
 		for j, p := range e.Parameters {
-			params[j] = p.Parameter
+			params[j] = p
+			if err := collectNamedTypes(p.ExtendedType, namedTypes, o.CollectedNamedTypes, di.MainPkg); err != nil {
+				return nil, err
+			}
+			if p.ExtendedType != nil {
+				params[j].ExtendedType = &manifest.ExtendedType{}
+				buildManifestExtendedType(params[j].ExtendedType, p.ExtendedType)
+			}
```

Comment:
If `p.ExtendedType` is not nil, then `params[j].ExtendedType` is set to exactly the same type. We need to build `p.ExtendedType` properly from the beginning in order not to spawn additional convertors here. What's the problem with having proper `p.ExtendedType`?